test/integration/integration_test.go

```diff
@@ -217,28 +218,60 @@ func ensureTestDirEmpty() {
 }
 
 //  To get values of nested keys, use following dot formating in Scenarios: key.nestedKey
-//  If an array is expected, use following formating: [value1 value2 value3].
-func configContains(configPath, expectingResult, expectedKeyPath, expectedValue string) error {
+//  If an array is expected, then expect: "[value1 value2 value3]"
+//  If empty string, non existing value are expected, then expect "<nil>"
+func getConfigValue(configPath string, keyPath string) (string, error) {
+	var keyValue string
 	data, err := ioutil.ReadFile(testDir + "/" + configPath)
 	if err != nil {
-		return fmt.Errorf("Cannot read config file: %s", err)
+		return "", fmt.Errorf("Cannot read config file: %v", err)
 	}
 	var values map[string]interface{}
 	json.Unmarshal(data, &values)
-	actualValue := ""
-	keyPath := strings.Split(expectedKeyPath, ".")
-	for _, element := range keyPath {
+	keyPathArray := strings.Split(keyPath, ".")
+	for _, element := range keyPathArray {
 		switch value := values[element].(type) {
 		case map[string]interface{}:
 			values = value
 		case []interface{}, nil, string, float64:
-			actualValue = fmt.Sprintf("%v", value)
+			keyValue = fmt.Sprintf("%v", value)
 		default:
-			return errors.New("Unexpected type in JSON config, not supported by testsuite")
+			return "", errors.New("Unexpected type in JSON config, not supported by testsuite")
 		}
 	}
-	if (expectingResult == "contains") && (actualValue != expectedValue) {
-		return fmt.Errorf("For key '%s' config contains unexpected value '%s'", expectedKeyPath, actualValue)
+	return keyValue, nil
+}
+
+func jsonConfigContains(configPath string, condition string, keyPath string, expectedValue string) error {
+	expectedRegexp, err := regexp.Compile(expectedValue)
+	if err != nil {
+		return fmt.Errorf("Expected value must be a valid regular expression statement: ", err)
+	}
+
+	keyValue, err := getConfigValue(configPath, keyPath)
+	if err != nil {
+		return err
+	}
+
+	if (condition == "contains") && !expectedRegexp.MatchString(keyValue) {
+		return fmt.Errorf("For key '%s' config contains unexpected value '%s'", keyPath, keyValue)
+	} else if (condition == "does not contain") && expectedRegexp.MatchString(keyValue) {
+		return fmt.Errorf("For key '%s' config contains value '%s', which it should not contain", keyPath, keyValue)
+	}
+
+	return nil
+}
+
+func jsonConfigHasKey(configPath string, condition string, keyPath string) error {
```

Comment:
The name is actually misleading. It implies that it checks whether the config defines a given key. However, the logic of the method depends on _contdition_. The same applies for _ jsonConfigContains_. Maybe more generic function names would be better, _checkConfigKey_ and _matchConfigValue_?